### ISSUE TITLE
[ FEATURE REQUEST]: Add an option to customize the background color of dock #121 

### ISSUE THREAD

**Issue by brymer-meneses** [pop-os/cosmic-dock#28]
As far as I'm aware, there's currently no option to customize the background color of the dock.

**Comment by NathanaelA** [comment]
Yeah, it appears they were trying to make it "simpler", but this creates the inability to make certain changes as the default settings UI is no longer present.      You can use "dconf-editor" and then navigate to 'org.gnome.shell.extensions.dash-to-dock' and then you can see all the settings.    One of them is `custom background color`.

**Comment by brymer-meneses** [comment]
Thank you @NathanaelA !

**Comment by icristi2** [comment]
Has someone actually found a way to change the dock background color (Cosmic Dock) in Pop OS 21.04?
I've tried via dconf Editor without any success.
Modifed:
  `custom-background-color` to TRUE
  `background-color` to a custom value: #282A36 in this case, as I wanted to match my Dracula theme
in **org.gnome.shell.extensions.dash-to-dock**
but no changes seem to be made.
...which is kind of strange, as modifying other parameters work, like: `require-pressure-to-show`, `running-indicator-dominant-color,` `show-trash`.
Is there any other setting/parameter that needs to be modified, or there are some bugs?

**Comment by brymer-meneses** [comment]
I just turned off cosmic-dock and installed floating-dock through the gnome extensions website. Floating-dock allows the customization of the background color, so I sticked with it.

**Comment by icristi2** [comment]
I guess, until they fix this / make it consistent (at least to be able to change via dconf) or add it as an option in UI using other docks is the only solution.
I'll keep investigating, but I guess it's an issue on System76 end.
Thanks.

**Comment by icristi2** [comment]
What is interesting to know is if a SVG image is used for the Cosmic Dock background. Than it's understandable that tweaking `custom-background-color` would not work. I have briefly examined the files & code for Cosmic Dock but couldn't tell so far. Maybe someone more knowledgable can enlighten here.
But, what bothers me is that changing `apply-custom-theme` to TRUE with **dconf Editor** and setting the DASHBOARD background color through **gnome-shell.css** from my custom theme (Dracula in my case) still has no effect on background color; only app indicators are changed accordingly.

As a side note, by modifying **gnome-shell.css** (of user theme Dracula) I managed to get rid of that ugly **Applications** grey background ;)

**Comment by dmccollough1** [comment]
Glad to see I'm not the only one with this. It was my biggest issue in the beta and it looks like it persists in the current run. I've even tried using regular dash to dock and adjusting it to no avail.

**Comment by Neyl-123** [comment]
Why is it so complicated to change the dock color?
I don't have the dconf entries `custom-background-color` and `background-color`  in  org.gnome.shell.extensions.dash-to-dock and also changing the hex color in **gnome-shell.css** in /usr/share/themes/gnome-shell and ~/.themes/gnome-shell (of the Pop theme) does not work.

**Comment by ForsetGump1952** [comment]
to change the dock...
From Terminal run:
gsettings set org.gnome.shell.extensions.dash-to-dock background-opacity 0.8
gsettings set org.gnome.shell.extensions.dash-to-dock background-color '#2d3749'

you can also change the background colour of the applications menu by modifying the file dark.css file in User/Share/gnome-shell.extensions/pop-cosmic@system76.com. just change the 2nd line where it lists the background-color

**Comment by Neyl-123** [comment]
Thank you @ForsetGump1952, it finally worked

**Comment by Itz-Abhishek-Tiwari** [comment]
👋 Hey there! If you're looking for an easy way to customize your dock and application menu, this script might just be what you need!
https://github.com/Itz-Abhishek-Tiwari/pop-os-dock-application-color-changer
